### PR TITLE
feat(hbase): add hbase-entrypoint.sh script

### PR DESCRIPTION
**Pull request by razvan**
# Description

prere-quisite for : https://github.com/stackabletech/hbase-operator/pull/570

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Integration tests ran successfully
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

File: hbase/stackable/bin/hbase-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+handle_term_signal() {
+  if [ "${term_child_pid}" ]; then
+    if [ "regionserver" == "${HBASE_ROLE_NAME}" ]; then
+      echo "Start pre-shutdown"
+      # REGION_MOVER_OPTS is a string that contains multiple arguments and needs to be spliced here
+      # therefore disable shellcheck for this line
+      # shellcheck disable=SC2086
+      /stackable/hbase/bin/hbase org.apache.hadoop.hbase.util.RegionMover ${REGION_MOVER_OPTS}
```
Comment (NickLarsenNZ) [inline hbase/stackable/bin/hbase-entrypoint.sh:38]: There is a valid way to do it without disabling the shell check, but I can't recall it right now.

Nothing to do here though, if I find it I will mention it.

File: hbase/stackable/bin/hbase-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+wait_for_termination() {
+  set +e
+  term_child_pid=$1
+  if [[ -v term_kill_needed ]]; then
+    kill -TERM "${term_child_pid}" 2>/dev/null
+  fi
+  wait "${term_child_pid}" 2>/dev/null
+  trap - TERM
```
Comment (NickLarsenNZ) [inline hbase/stackable/bin/hbase-entrypoint.sh:54]: What does this do? Run nothing on SIGTERM?

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

File: hbase/stackable/bin/hbase-entrypoint.sh (hunk 2 above)
Comment (razvan) [reply hbase/stackable/bin/hbase-entrypoint.sh:54]: [Copy&pasta from op-rs.](https://github.com/stackabletech/operator-rs/blob/82a0ef6d14c6f4e144ee1b4f5eff6abebf0108f2/crates/stackable-operator/src/utils/bash.rs#L40) 

I don't quite get understand it either.

**Review by NickLarsenNZ — APPROVED** [review]
Approving so we aren't blocked, but would like some light docs on the traps/signal handlin.

**Comment by lfrancke** [comment]
@razvan would we revert this if we decide not to merge the other related graceful shutdown tickets?

**Comment by razvan** [comment]
It's not used and has no impact if it stays.

Personally I would prefer to create entry point scripts for all products and get rid of the hard-coded shell snippets in the operators.

But if it bothers you, I can revert it.

**Comment by lfrancke** [comment]
It doesn't. I was just wondering.

**Pull request merged**